### PR TITLE
Add support for configuring extra CIDRs for Control Plane SG

### PR DESCRIPTION
**Pull request by MrBones757**
### Description

See: https://github.com/weaveworks/eksctl/issues/772

When creating a private cluster from a network outside of the VPC, provisioning fails.
This PR makes use of the existing extraCIDRs fields in the vpc config of cluster yaml to append to
the control plane security group ingress rules.

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by js-rwwa** [comment]
I have helped with these changes/tests - commenting to follow

**Comment by MrBones757** [comment]
Added basic unit tests in addition to the logic for adding extra CIDRs to Control Plane SG (ingress)
Let me know if you require anything else for this PR.

Tested Deploy Operation - Success (working internally after public access disabled)
Tested Update Operation - Does not work, i suspect because of " NOTE: cluster VPC (subnets, routing & NAT Gateway) configuration changes are not yet implemented". SG config logic was changed in VPC, so by extension does this mean that if an extraCIDRs update was needed it should be done when VPC updates are supported?

**Comment by Callisto13** [comment]
Thanks for opening this PR @MrBones757, the team will review shortly 👍 

> Tested Update Operation - Does not work

Do you mean the `eksctl upgrade cluster` command? Upgrade _only_ bumps the control plane version. Eksctl currently provides no way of up_dating_ any cluster configuration.

**Comment by MrBones757** [comment]
awesome, yes this is what i was referring to.

I was jumping though the code today in debug mode and could see a few places where I taught I might be able to implement such a configuration update method for extraCIDRs, though given that update functionality is not in scope at the moment for eksctl ill leave it at what i've done here so far, unless there is anything else you and your team can see that needs doing :)

Would be happy to help adding that functionality though at another time

**Comment by MrBones757** [comment]
I believe i have fixed the lint errors, though im not 100% sure why the test-and-build unit tests are failing / how to resolve that one.

I see the following in the output:
HINT: to fix this, run 'git commit pkg/nodebootstrap/bindata/assets.go pkg/addons/default/assets.go pkg/addons/assets.go pkg/apis/eksctl.io/v1alpha5 userdocs/src/usage/schema.json pkg/apis/eksctl.io/v1alpha5/zz_generated.deepcopy.go pkg/eks/mocks/EKSAPI.go pkg/eks/mocks/ELBAPI.go pkg/eks/mocks/CloudFormationAPI.go pkg/eks/mocks/IAMAPI.go pkg/eks/mocks/EC2API.go pkg/eks/mocks/AutoScalingAPI.go pkg/eks/mocks/CloudTrailAPI.go pkg/eks/mocks/STSAPI.go pkg/eks/mocks/ELBV2API.go pkg/eks/mocks/SSMAPI.go --message "Update generated files"'

I made a change to zz_generated.deepcopy.go, as i wasnt sure how to re-generate this myself (windows) though i made the change mirror the existing cidrs []string field that was there

**Comment by Callisto13** [comment]
Ah yeh this is an annoying but necessary thing. Basically since you have changed a file which will cause some other autogenerated files to update, you need to run the generate and commit these _before_ you run the tests. It's before my time, but I believe this catch was put in place so that when CI run the generate (as part of `make test`, it doesn't just throw new files "into the void", but instead yells and makes sure they are committed.

So TLDR:
- `make test` will run generate first and fail if changes are detected and uncommitted
- commit those files
- run `make test` again, should be good

**Comment by MrBones757** [comment]
Thanks for getting back to me on this - working on the above this morning.
As part of this PR (or in a separate PR) is it worth me updating the readme / contrib guide to include some of these things? 
Such as: doco about the code generation and pitfalls using this for the first time & how to resolve (basically what you said above), additional pitfalls i found working on windows (make not workign after code has been opened in windows, WSL instructions for using make etc)

**Comment by MrBones757** [comment]
i believe all the files that needed fixing via the make test should be good now :) tests appear to succeed locally (switched to my WSL env - much better now)

**Comment by Callisto13** [comment]
> As part of this PR (or in a separate PR) is it worth me updating the readme / contrib guide to include some of these things?
Such as: doco about the code generation and pitfalls using this for the first time & how to resolve (basically what you said above), additional pitfalls i found working on windows (make not working after code has been opened in windows, WSL instructions for using make etc)

🙌  This would be AWESOME (in a separate PR). The note about the code generation would fit as part of [this list](https://github.com/weaveworks/eksctl/blob/main/CONTRIBUTING.md#writing-your-solution) I think, and a section about Windows dev gotchas would be good anywhere under [here](https://github.com/weaveworks/eksctl/blob/main/CONTRIBUTING.md#developing-eksctl). None of the core team use Windows, so we naturally don't come across issues there: anything you can add would be sweet 🙏

**Review by Callisto13 — APPROVED** [review]
LGTM!

**Comment by cdenneen** [comment]
This needs to be added to the vpc-configuration.md and examples for clear documentation of how this feature is to be used. Currently it still seems like this feature is missing rather than available.

**Comment by Callisto13** [comment]
Could you open a new issue to request that? It is unlikely to be picked up here
